longhorndev/dev.go

```diff
@@ -404,16 +404,45 @@ func (d *LonghornDevice) Expand(size int64) error {
 	d.Lock()
 	defer d.Unlock()
 
-	if d.scsiDevice != nil {
-		return fmt.Errorf("cannot expand the device %v to size %v since the frontend %v is already up", d.name, size, d.frontend)
-	}
-
 	if d.size > size {
 		return fmt.Errorf("device %v: cannot expand the device from size %v to a smaller size %v", d.name, d.size, size)
 	} else if d.size == size {
 		return nil
 	}
 	d.size = size
 
+	if d.scsiDevice == nil {
+		return nil
```

Comment:
Suggest returning error, because this is not a valid case.

Reply:
Nope... This means the frontend is disabled. It's still a valid case and the expansion function doesn't need to do anything for it.

Reply:
But I will add a info log anyway.

Reply:
yeah, it's fair.